apps/builder/app/builder/features/settings-panel/variable-popover.tsx

```diff
@@ -360,17 +368,48 @@ const VariablePanel = forwardRef<
     }
     return "string";
   });
+  const typeOptions: VariableType[] = [
+    "string",
+    "number",
+    "boolean",
+    "json",
+    "resource",
+  ];
+  const getTypeLabel = (value: VariableType) =>
+    value === "json" ? "JSON" : humanizeString(value);
   const typeFieldElement = (
     <Flex direction="column" css={{ gap: theme.spacing[3] }}>
       <Label>Type</Label>
       <Select<VariableType>
-        options={["string", "number", "boolean", "json", "resource"]}
-        getLabel={(value: VariableType) =>
-          value === "json" ? "JSON" : humanizeString(value)
-        }
+        options={typeOptions}
+        getLabel={getTypeLabel}
         value={type}
         onChange={setType}
-      />
+      >
+        {typeOptions.map((option) =>
+          option === "resource" && allowResourceVariables === false ? (
```

Comment:
Nit: I keep single SelectItem and just set disabled and badge conditionally, this notation was hard to read

Reply:
So you want this condition to be written twice instead?